File: Python/4_Pre_Processing/Test_Train/Test_Train.py
#########################################################################################################
# Name             : Test and Train
# Date             : 04-21-2015
# Author           : Christopher M
# Dept             : BEI Business Analytics
#########################################################################################################
# ver    user        date(YYYYMMDD)        change  
# 1.0    CMooney      20150421             initial
#########################################################################################################

import pandas as pd
from sklearn.cross_validation import train_test_split

df = pandas.read_csv('F:\\Analytics_Process\\R\\SampleData\\iris_missing_Values.csv')
print(df.head(10))

Matrix = df.values
print(Matrix.shape)   
 
# Select our features (predictors)

# if our target is the last vector we can use this simpler method
MatrixFeatures = Matrix[:,:-1]

# if our target is not the we specify the features
MatrixFeatures = Matrix[:,[0,1,2,3]]


# Select our target is the last last vector we can simply say only return last vector
MatrixTarget = Matrix[:,[-1]]

# if our target is not the we specify the index
MatrixTarget = Matrix[:,[4]]

# Now we can split our data into test and train
TrainFeatures, TestFeatures, TrainTargets, TestTargets = train_test_split(MatrixFeatures, MatrixTarget, test_size=0.4)
    
# Training Sets
print(TrainFeatures.shape)   
print(TrainTargets.shape)   

# Testing Sets
print(TestFeatures.shape)   
print(TestTargets.shape)   

File: R/4_Pre_Processing/Test_Train/TestTrain.r
###############################################################################################################################
# Name             : TestTrain
# Date             : 07-15-2015
# Author           : Christopher M
# Dept             : Business Analytics
# Purpose          : Examples of how to split into test and train
# Called by        : Not in production
###############################################################################################################################
# ver    user        date(YYYYMMDD)        change  
# 1.0    w47593      20150715             initial
###############################################################################################################################

##_____________________________________________________________________________________________________________________________
# Load Packages

library(caret)

##_____________________________________________________________________________________________________________________________
# Split into test and train using Caret Package

trainIndex <- createDataPartition(iris$Species, p = .6,
                                  list = FALSE,
                                  times = 1)
irisTrain <- iris[ trainIndex,]
irisTest  <- iris[-trainIndex,]

##_____________________________________________________________________________________________________________________________
# Split into test and train using sqldf Package

SplitPercent = .6
NumberRecords = .6*nrow(iris)

iris$index = rownames(iris)
query = paste("Select * from iris order by random() limit " ,NumberRecords,sep='')
irisTrain = sqldf(query) 

# index is a reserved word therefore we wrap it in brackets
query = paste("Select * from iris where [index] not in (select [index] from irisTrain)")
irisTest = sqldf(query) 